Assignment_1_Pomodoro/app/src/main/java/no/uia/ikt205/pomodoro/MainActivity.kt

```diff
@@ -13,28 +13,69 @@ class MainActivity : AppCompatActivity() {
 
     lateinit var timer:CountDownTimer
     lateinit var startButton:Button
+    lateinit var thirtyMinButton:Button
+    lateinit var sixtyMinButton:Button
+    lateinit var ninetyMinButton:Button
+    lateinit var twoHourButton:Button
     lateinit var coutdownDisplay:TextView
 
-    val timeToCountDownInMs = 5000L
+    var timeToCountDownInMs = 5000L
     val timeTicks = 1000L
+    var isCounting = false
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
        startButton = findViewById<Button>(R.id.startCountdownButton)
        startButton.setOnClickListener(){
-           startCountDown(it)
+           if(!isCounting){
+               startCountDown(it)
+           }
+           isCounting = true
        }
+       thirtyMinButton = findViewById<Button>(R.id.startThirtyMinCountdown)
+       thirtyMinButton.setOnClickListener(){
+           if (isCounting){
+                resetTimer()
+           }
+           timeToCountDownInMs = 1800000L
+           updateCountDownDisplay(timeToCountDownInMs)
+       }
```

Comment:
A very smart way to stop the timer if you pressed the wrong button, great job Thomas 👍